### PR TITLE
modified swipeTo

### PR DESCRIPTION
**Pull request by ashby-bowles**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
     <groupId>com.github.willowtreeapps</groupId>
     <artifactId>conductor-mobile</artifactId>
-    <version>0.8.0</version>
+    <version>0.8.5</version>
```
Comment (konwinkler) [inline pom.xml]: Good job keeping this on the correct working version. This really helps managing releases!

File: src/main/java/com/joss/conductor/mobile/Locomotive.java [hunk 2]
```diff
@@ -572,19 +572,23 @@ private Locomotive performCornerSwipe(ScreenCorner corner, SwipeElementDirection
     }
 
     public WebElement swipeTo(SwipeElementDirection s, By by, int attempts) {
```
Comment (konwinkler) [inline src/main/java/com/joss/conductor/mobile/Locomotive.java]: can we change the name of `s` to make it more meaningful? how about `direction`

File: src/main/java/com/joss/conductor/mobile/Locomotive.java [hunk 3]
```diff
@@ -572,19 +572,23 @@ private Locomotive performCornerSwipe(ScreenCorner corner, SwipeElementDirection
...
+            swipeCenterLong(s);
+            try {
+                element = driver.findElement(by);
+                // element was found, check for visibility
+                if (element.isDisplayed()) {
+                    // element is in view, exit the loop
+                    return element;
                 }
+                // element was not visible, continue scrolling
+            } catch (org.openqa.selenium.NoSuchElementException e) {
```
Comment (konwinkler) [inline src/main/java/com/joss/conductor/mobile/Locomotive.java]: can we do an import at the beginning of the file for the exception? `import org.openqa.selenium.NoSuchElementException`?
Also let's not abbreviate variable, lets change `e` to `exception` or `ignored`

File: src/main/java/com/joss/conductor/mobile/Locomotive.java [hunk 4]
```diff
@@ -572,19 +572,23 @@ private Locomotive performCornerSwipe(ScreenCorner corner, SwipeElementDirection
...
+                // element could not be found, continue scrolling
             }
         }
-        Logger.error("WARN: Element" + by.toString() + "does not exist!");
+        // element could not be found or was not visible, return null
+        System.err.println("WARN: Element" + by.toString() + "does not exist!");
```
Comment (konwinkler) [inline src/main/java/com/joss/conductor/mobile/Locomotive.java]: Let's use the logging library so we can turn it off and on. Also the `WARN:` should be generated by the correct level, so `Logger.warn("Element" + by.toString() + "does not exist!");`

File: src/main/java/com/joss/conductor/mobile/Locomotive.java [hunk 5]
```diff
@@ -571,20 +570,24 @@ private Locomotive performCornerSwipe(ScreenCorner corner, SwipeElementDirection
...
+                    return element;
                 }
+                // element was not visible, continue scrolling
+            } catch (NoSuchElementException exception) {
+                // element could not be found, continue scrolling
             }
         }
-        Logger.error("WARN: Element" + by.toString() + "does not exist!");
+        // element could not be found or was not visible, return null
+        Logger.warn("WARN: Element" + by.toString() + "does not exist!");
```
Comment (konwinkler) [inline src/main/java/com/joss/conductor/mobile/Locomotive.java]: you don't need `WARN: ` in the string

**Comment by fuzzybinary** [comment]
I agree with all @konwinkler's suggestions. Also please write a clearer commit message - what did you modify and why?

**Comment by ashby-bowles** [comment]
ok @fuzzybinary  @konwinkler should be good now

**Pull request merged**